packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -126,6 +133,47 @@ private CameraPosition getCameraPosition() {
     return trackCameraPosition ? googleMap.getCameraPosition() : null;
   }
 
+  /**
+   * Invalidates the map view after the map has finished rendering.
+   *
+   * <p>gmscore GL renderer uses a {@link android.view.TextureView}. Android platform views that are
+   * displayed as a texture after Flutter v3.0.0. require that the view hierarchy is notified after
+   * all drawing operations have been flushed.
+   *
+   * <p>Since the GL renderer doesn't use standard Android views, and instead uses GL directly, we
+   * notify the view hierarchy by invalidating the view.
+   *
+   * <p>Unfortunately, when {@link GoogleMap.OnMapLoadedCallback} is fired, the texture may not have
+   * been updated yet.
+   *
+   * <p>To workaround this limitation, wait two frames. This ensures that at least the frame budget
+   * (16.66ms at 60hz) have passed since the drawing operation was issued.
+   */
+  private void invalidateMapIfNeeded() {
+    if (googleMap == null || invalidatePending) {
```

Comment:
Don't we actually need to restart the callback counter if a new invalidate request comes in? Otherwise we won't actually wait 2 frames after the last invalidation.

Reply:
I changed it to `loadedCallbackPending`. Note that `setOnMapLoadedCallback` is fired when the map has finished rendering, so gmscore has a batching mechanism.